DEPENDENCIES/simple_ml_helib/src/HelibCkksContext.cpp

```diff
@@ -45,7 +45,7 @@ void HelibCkksContext::init(unsigned long m,unsigned long r,unsigned long L) {
 
 void HelibCkksContext::init(const HelibConfig& conf) {
   config=conf;
-  unsigned long p = -1;
+  unsigned long p = std::numeric_limits<unsigned long>::max();
```

Comment:
This needs to be solved in another way. In HElib `p` (at least in some places) is `long`, and `-1` is a special value which chooses CKKS scheme. We need all the arguments leading there to be `long` as well.

If you're not sure how to fix (I haven't looked at this yet) you can drop this specific change from the PR and submit without it.

Reply:
Temporary removed affected commits, thanks for analysis!